src/pages/Synchronization/UserSyncTest/helper.js

```diff
@@ -0,0 +1,250 @@
+import without from 'lodash/without'
+import {
+    apiFetchOU,
+    apiFetchOUSearch,
+    apiFetchProgramRulesBasic,
+} from './queries/syncQueries'
+import { getMetadataSize } from './getMetadata'
+import { getDataSize } from './getData'
+
+export const createInitialValues = initialValues => ({
+    organisationUnitsNumber: initialValues.organisationUnitsNumber || 0,
+    organisationUnitSearchNumber:
+        initialValues.organisationUnitSearchNumber || 0,
+    datasetNumber: initialValues.datasetNumber || 0,
+    programNumber: initialValues.programNumber || 0,
+    programRuleNumber: initialValues.programRuleNumber || 0,
+    reservedValueNumber: initialValues.reservedValueNumber || 0,
+    metadataSize: initialValues.metadataSize || 0,
+    dataSize: initialValues.dataSize || 0,
+})
+
+/**
+ * run user test
+ * get elements and metadata and data download size
+ * */
+export const runUserTest = async ({
+    user,
+    dataEngine,
+    reservedValues,
+    globalSettings,
+    specificSettings,
+}) => {
+    let orgUnitSearch = 0
+    let orgUnit = {}
+    let programs = {}
+    let dataSets = {}
+    let programRules = {}
+    let metadataSize = 0
+    let dataSize = 0
+
+    await getTestElements(user, dataEngine).then(result => {
+        orgUnitSearch = result.orgUnitSearch
+        orgUnit = result.orgUnit
+        programs = result.program
+        dataSets = result.dataSet
+        programRules = result.programRule
+    })
+
+    await getMetadataSize({
+        dataEngine,
+        orgUnitList: orgUnit.idList,
+        programList: programs.idList,
+        trackedEntityTypeList: programs.trackedEntityTypeList,
+        teaList: programs.trackedEntityAttribute,
+        optionSetList: programs.optionSet,
+        dataSetList: dataSets.idList,
+        dataElementList: dataSets.dataElement,
+        categoryComboList: dataSets.categoryCombo,
+        categoryList: dataSets.category,
+        indicatorList: dataSets.indicator,
+    }).then(result => (metadataSize = result))
+
+    await getDataSize({
+        dataEngine,
+        orgUnit: orgUnit.idList,
+        program: specificSettings,
+        globalLimit: globalSettings,
+    }).then(result => (dataSize = result))
+
+    return {
+        organisationUnitsNumber: orgUnit.total,
+        organisationUnitSearchNumber: orgUnitSearch,
+        datasetNumber: dataSets.total,
+        programNumber: programs.total,
+        programRuleNumber: programRules.total,
+        reservedValueNumber: reservedValues,
+        metadataSize,
+        dataSize,
+    }
+}
+
+const getSearchOrgUnit = (orgUnits, dataEngine) => {
+    const orgUnitPromises = []
+    const orgUnitSearch = []
+    orgUnits.map(orgUnit =>
+        orgUnitPromises.push(apiFetchOUSearch(dataEngine, orgUnit.id))
+    )
+
+    return Promise.all(orgUnitPromises).then(result => {
+        result.map(ouList => ouList.map(ou => orgUnitSearch.push(ou)))
+        return uniqBy(orgUnitSearch, 'id').length
+    })
```

Comment:
Better to do this sequentially:

```suggestion
    const ouLists = Promise.all(orgUnitPromises)
    ouLists.forEach(ouList => 
        ouList.forEach(ou => {
            orgUnitSearch.push(ou)
        })
    }
    return uniqBy(orgUnitSearch, 'id').length
```